### PR TITLE
Jpeg color space deduction fix

### PR DESCRIPTION
**Pull request by br3aker**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description
Fixes https://github.com/SixLabors/ImageSharp/issues/2174

### CODE REVIEW

**Comment by brianpopow** [comment]
There were 3 failing tiff tests, but I believe it was an issue with the Tiff JpegDecompressor. Hopefully this 1959bf65 fixes it now.

**Comment by br3aker** [comment]
> There were 3 failing tiff tests, but I believe it was an issue with the Tiff JpegDecompressor. 

I believe it's not not that simple :D

TIFF with PhotometricInterpretation=RGB will have YCbCr signature at jpeg decoding level which would lead to invalid results.
Always using Rgb24 converter would lead to YCbCr encoded jpeg having RGB color conversion. I guess I have an idea: we can just pass PhotometricInterpretation to new `TiffSpectralConverter` and resolve color converter using that value.

File: src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs [hunk 1]
```diff
@@ -506,90 +506,48 @@ public void Dispose()
         /// Returns the correct colorspace based on the image component count and the jpeg frame component id's.
```
Comment (brianpopow) [inline src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs:506]: The comment is no longer valid, maybe change it like this:
```suggestion
        /// Returns the correct colorspace based on the image component count and adobe APP14 marker's color transform flag.
```

File: src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs [hunk 2]
```diff
@@ -506,90 +506,48 @@ public void Dispose()
...
         /// </summary>
         /// <param name="componentCount">The number of components.</param>
+        /// <param name="adobeMarker">Parsed adobe APP14 marker.</param>
         /// <returns>The <see cref="JpegColorSpace"/></returns>
-        private JpegColorSpace DeduceJpegColorSpace(byte componentCount)
+        internal static JpegColorSpace DeduceJpegColorSpace(byte componentCount, ref AdobeMarker adobeMarker)
         {
-            if (componentCount == 1)
```
Comment (brianpopow) [inline src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs:512]: I have found a jpeg image with an app14 marker which is gray, should that not be possible according to the spec?
gray-sample

Comment (br3aker) [reply src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs:512]: Adobe spec doesn't say anything about grayscale images but I think we can include it as decoding it doesn't violate the spec :)

**Comment by br3aker** [comment]
I've implemented tests for the specific method which resolves encoded jpeg color space. Not only it's easier to maintain, it is also much faster and there's no need to find a ton of images with different setups.

File: tests/ImageSharp.Tests/Formats/Jpg/JpegDecoderTests.Internal.cs [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    [Trait("Format", "Jpg")]
+    public partial class JpegDecoderTests
+    {
+        [Theory]
+        [InlineData(1, 0, JpegColorSpace.Grayscale)]
+        [InlineData(3, JpegConstants.Adobe.ColorTransformUnknown, JpegColorSpace.RGB)]
+        [InlineData(3, JpegConstants.Adobe.ColorTransformYCbCr, JpegColorSpace.YCbCr)]
+        [InlineData(4, JpegConstants.Adobe.ColorTransformUnknown, JpegColorSpace.Cmyk)]
+        [InlineData(4, JpegConstants.Adobe.ColorTransformYcck, JpegColorSpace.Ycck)]
+        internal void DeduceJpegColorSpaceAdobeMarker_ShouldReturnValidColorSpace(byte componentCount, byte adobeFlag, JpegColorSpace expectedColorSpace)
```
Comment (JimBobSquarePants) [inline tests/ImageSharp.Tests/Formats/Jpg/JpegDecoderTests.Internal.cs:33]: This is all very nice 👍

**Review by JimBobSquarePants — APPROVED** [review]
This all looks great! Thanks @br3aker

**Pull request merged**